test/generaloption.py

```diff
@@ -838,14 +847,27 @@ def test_option_as_value(self):
         self._match_testoption1_sysexit(['-b', '-s'],
                                         "-s option requires an argument")
 
-        # valid ways of specifying '-b' as a value to --store/-s
+        # only valid way of specifying '-b' as a value to --store
         topt = TestOption1(go_args=['--store=-b'])
         self.assertEqual(topt.options.store, '-b')
         self.assertFalse(topt.options.base)  # remain False (default value)
 
-        topt = TestOption1(go_args=['-s-b'])  # equivalent to --store=-b
-        self.assertEqual(topt.options.store, '-b')
-        self.assertFalse(topt.options.base)  # remain False (default value)
+        # when -b/--base is an option, the following are not accepted, since they're likely not what intended
```

Comment:
add tests with `ALLOW_OPTION_NAME_AS_VALUE` explicitly set to True or False, and test the default value